components/UsMap.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import statesLived from "../public/data/stateslived.json"
+import stateScores from "../public/data/state-scores.json"
+import usData from "../public/data/us-states.json"
+import * as d3 from "d3"
+
+
+class UsMap extends Component {
+
+  componentDidMount() {
+
+
+    this.statesLived = statesLived;
+    this.stateScores = stateScores;
+    this.usData = usData;
+    this.height = 500; (64, 0, 64)
+    this.width = 960;
+    this.colorRange = [
+      // lightest
+      "rgb(168,0,168)",
+      "rgb(128,0,128)",
+      "rgb(98,0,98)",
+      "rgb(78,0,78)",
+      "rgb(58,0,58)"
+    ]
+    this.renderMap();
+  }
+
+  mapScoresToStates(scoreData, usData) {
+
+    for (var i = 0; i < scoreData.length; i++) {
+      let scoreStateName = scoreData[i].state;
+      var stateScores = scoreData[i].score;
+      for (var j = 0; j < usData.features.length; j++) {
+        var jsonStateName = usData.features[j].properties.name;
+        if (scoreStateName == jsonStateName) {
+          usData.features[j].properties.score = stateScores;
+          break;
+        }
+      }
+    }
+    return usData
+  }
+
+  renderMap() {
+
+    // just unpacking for tidier variable names downstream
+    let usData = this.usData;
```

Comment:
No change needed, but you could write this as destructured like:

`let { usData, statesLived, colorRagen, width, height, stateScores } = this;`

Reply:
Oh, that's nice. I'm new to JS - never seen this, ty.